api/v1beta1/workspace_validation.go

```diff
@@ -319,80 +326,87 @@ func (r *ResourceSpec) validateCreateWithInference(inference *InferenceSpec, byp
 			return errs
 		}
 	}
+
 	instanceType := string(r.InstanceType)
 
-	skuHandler, err := utils.GetSKUHandler()
-	if err != nil {
-		errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("Failed to get SKU handler: %v", err), "instanceType"))
-		return errs
-	}
+	if featuregates.FeatureGates[consts.FeatureFlagDisableNodeAutoProvisioning] {
+		if presetName != "" { // TODO: can preset name ever be empty?
+			errs = errs.Also(r.validateBYONodeSelection(presetName))
+		}
+	} else {
+		skuHandler, err := utils.GetSKUHandler()
+		if err != nil {
+			errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("Failed to get SKU handler: %v", err), "instanceType"))
+			return errs
+		}
 
-	// Check if instancetype exists in our SKUs map for the particular cloud provider
-	if skuConfig := skuHandler.GetGPUConfigBySKU(instanceType); skuConfig != nil {
-		if runtime != model.RuntimeNameVLLM {
-			if presetName != "" {
-				modelPreset := plugin.KaitoModelRegister.MustGet(presetName) // InferenceSpec has been validated so the name is valid.
-				params := modelPreset.GetInferenceParameters()
-
-				machineCount := *r.Count
-				machineTotalNumGPUs := resource.NewQuantity(int64(machineCount*skuConfig.GPUCount), resource.DecimalSI)
-				machineTotalGPUMem := resource.NewQuantity(int64(machineCount*skuConfig.GPUMemGB)*consts.GiBToBytes, resource.BinarySI) // Total GPU memory
-
-				modelGPUCount := resource.MustParse(params.GPUCountRequirement)
-				modelTotalGPUMemory := resource.MustParse(params.TotalSafeTensorFileSize)
-
-				// Separate the checks for specific error messages
-				if machineTotalNumGPUs.Cmp(modelGPUCount) < 0 {
-					if bypassResourceChecks {
-						klog.Warningf("Bypassing resource check: Insufficient number of GPUs detected but continuing due to bypass flag. Instance type %s provides %s, but preset %s requires at least %d",
-							instanceType, machineTotalNumGPUs.String(), presetName, modelGPUCount.Value())
-					} else {
-						errs = errs.Also(apis.ErrInvalidValue(
-							fmt.Sprintf(
-								"Insufficient number of GPUs: Instance type %s provides %s, but preset %s requires at least %d",
-								instanceType,
-								machineTotalNumGPUs.String(),
-								presetName,
-								modelGPUCount.Value(),
-							),
-							"instanceType",
-						))
+		// Check if instancetype exists in our SKUs map for the particular cloud provider
+		if skuConfig := skuHandler.GetGPUConfigBySKU(instanceType); skuConfig != nil {
+			if runtime != model.RuntimeNameVLLM {
+				if presetName != "" {
```

Comment:
There's an existing check here, can preset name ever be empty or can we remove this?